worker/containerd/containerd.go

```diff
@@ -56,19 +58,24 @@ func newContainerd(root string, client *containerd.Client, snapshotterName strin
 		xlabels[k] = v
 	}
 
-	gc := func(ctx context.Context) error {
+	throttledGC := throttle.Throttle(time.Second, func() {
 		// TODO: how to avoid this?
+		ctx := context.TODO()
 		snapshotter := client.SnapshotService(snapshotterName)
 		ctx = namespaces.WithNamespace(ctx, "buildkit")
 		key := identity.NewID()
 		if _, err := snapshotter.Prepare(ctx, key, "", snapshots.WithLabels(map[string]string{
 			"containerd.io/gc.root": time.Now().UTC().Format(time.RFC3339Nano),
 		})); err != nil {
-			return err
+			logrus.Errorf("GC error: %+v", err)
 		}
 		if err := snapshotter.Remove(ctx, key); err != nil {
-			return err
+			logrus.Errorf("GC error: %+v", err)
 		}
+	})
+
+	gc := func(ctx context.Context) error {
```

Comment:
why break ctx?

Reply:
This context may be already canceled by the time the function gets called. The function in called on background in a different context.